drivers/ieee802154/ieee802154_nrf5.c

```diff
@@ -650,6 +688,69 @@ static void nrf5_iface_init(struct net_if *iface)
 	ieee802154_init(iface);
 }
 
+#if defined(CONFIG_NRF_802154_ENCRYPTION)
+static void nrf5_config_mac_keys(struct ieee802154_key *mac_keys)
+{
+	nrf_802154_security_error_t err;
+	nrf_802154_key_t key;
+	uint8_t key_id_to_remove;
+
+	__ASSERT(mac_keys, "Invalid argument.");
+
+	/* Remove old invalid key assuming that its index is first_valid_key_id - 1.
+	 * TODO: This is Thread specific assumption, need to be changed when RD will provided
+	 * API for removing all keys or handling this internally.
+	 */
+	key_id_to_remove = (*mac_keys).key_index == 1 ? 0x80 : (*mac_keys).key_index - 1;
+
+	key.id.mode = (*mac_keys).key_id_mode;
+	key.id.p_key_id = &key_id_to_remove;
+
+	nrf_802154_security_key_remove(&key.id);
+
+	while ((*mac_keys).key_value != NULL) {
+		key.value.p_cleartext_key = (*mac_keys).key_value;
+		key.id.mode = (*mac_keys).key_id_mode;
+		key.id.p_key_id = &(*mac_keys).key_index;
+		key.type = NRF_802154_KEY_CLEARTEXT;
+		key.frame_counter = 0;
+		key.use_global_frame_counter = !((*mac_keys).frame_counter_per_key);
+
+		nrf_802154_security_key_remove(&key.id);
+		err = nrf_802154_security_key_store(&key);
+		__ASSERT(err == NRF_802154_SECURITY_ERROR_NONE ||
+				 err == NRF_802154_SECURITY_ERROR_ALREADY_PRESENT,
+			 "Storing key failed, err: %d", err);
+		mac_keys++;
+	};
+}
+#endif /* CONFIG_NRF_802154_ENCRYPTION */
+
+#if defined(CONFIG_OPENTHREAD_CSL_RECEIVER)
```

Comment:
This `#if` - `#endif` block seems OpenThread specific. I think it should be replaced with something more generic like simple `nrf_802154_receive_at` abstraction and move most of this code to OpenThread radio platform implementation.